service/root.go

```diff
@@ -0,0 +1,60 @@
+package service
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/riandyrn/otelchi"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/zenledger-io/zazen/httpx"
+	"github.com/zenledger-io/zazen/telemetry"
+)
+
+var (
+	defaultRequestTimeout = 60 * time.Second
+)
+
+func newRoot(tp trace.TracerProvider, cfg Config) http.Handler {
+	root := chi.NewRouter()
+
+	timeout := cfg.RequestTimeout
+	if timeout == 0 {
+		timeout = defaultRequestTimeout
+	}
+
+	root.Use(middleware.Timeout(timeout))
+	root.Use(middleware.RealIP)
+	root.Use(middleware.RequestID)
```

Comment:
Curious what this sets it too, if the load balancers passes a request ID (not sure if aws does but heroku which is built on aws does) then it's better to use that so we can match up any logs.

Reply:
It uses the standard `X-Request-Id`: https://github.com/go-chi/chi/blob/master/middleware/request_id.go#L25 if its there, otherwise it creates one.